File: src/com/dafttech/terra/graphics/gui/elements/ElementLabel.java
package com.dafttech.terra.graphics.gui.elements;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.dafttech.terra.graphics.AbstractScreen;
import com.dafttech.terra.resources.Resources;
import com.dafttech.terra.world.Vector2;

public abstract class ElementLabel extends GUIElement {
    private CharSequence text = "Button";

    public ElementLabel(Vector2 p, String txt) {
        super(p, new Vector2(100, 20));

        text = txt;
    }

    @Override
    public void draw(AbstractScreen screen, Vector2 origin) {
        screen.batch.begin();
        Resources.BUTTON_FONT.draw(screen.batch, text, position.x + origin.x + size.x / 2 - Resources.BUTTON_FONT.getBounds(text).width / 2, 6 + position.y + origin.y);
        screen.batch.end();

        screen.batch.setColor(Color.WHITE);
    }
}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/elements/ElementLabel.scala
package com.dafttech.terra.engine.gui.elements

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.g2d.BitmapFont
import com.dafttech.terra.engine.AbstractScreen
import com.dafttech.terra.engine.renderer.TextBounds
import com.dafttech.terra.engine.vector.Vector2d
import com.dafttech.terra.resources.Resources

class ElementLabel(p: Vector2d, var text: String) extends GUIElement(p, null) {
  var clr: Color = Color.WHITE
  setText(text)

  def setText(txt: String): Unit = {
    text = txt
    size = TextBounds.getBounds(Resources.GUI_FONT, text)
  }

  def setColor(c: Color): Unit = {
    clr = c
  }

  override def draw(screen: AbstractScreen): Unit = {
    val p: Vector2d = getScreenPosition
    screen.batch.begin
    Resources.GUI_FONT.setColor(clr)
    Resources.GUI_FONT.draw(screen.batch, text, p.x.toFloat, 6 + p.y.toFloat)
    screen.batch.end
  }
}